### PR TITLE
Table "climbing" animation

### PR DESCRIPTION
**Pull request by alsoandanswer**
"Adds a small offset animation to climbing tables, so it doesn't look like you're teleporting onto them."

ported from bay, and it won't go weird when tables are flipped or dismantled.

### CODE REVIEW

**Comment by alsoandanswer** [comment]
!review

**Comment by skull132** [comment]
video please!

**Comment by Sindorman** [comment]
That isn't really climbing. Rather walking on higher elevation

**Comment by alsoandanswer** [comment]
> That isn't really climbing. Rather walking on higher elevation

Well, I wouldn't want it to be too visually intrusive. It just smoothens out the current one

File: code/modules/mob/animations.dm [hunk 1]
```diff
@@ -241,3 +240,17 @@ note dizziness decrements automatically in the mob's Life() proc.
...
+		if(pixel_z != check)
+			animate(src, pixel_z = check, time = 2, easing = SINE_EASING)
+	else
+		var/obj/structure/S = (locate() in get_turf(src))
+		if(!S && pixel_z != default_pixel_z)
+			animate(src, pixel_z = default_pixel_z, time = 2, easing = SINE_EASING)
+
+/mob/living/Move()
+	. = ..()
+	on_structure_offset(0)
```
Comment (Arrow768) [inline code/modules/mob/animations.dm]: As I mentioned on discord:

I do not like this.
This loops through the contents of every turf the mob moves into to check if there is a table.
Cant the reset be done with [Uncrossed?](http://www.byond.com/docs/ref/info.html#/atom/proc/Uncrossed)

**Comment by Geevies** [comment]
@alsoandanswer Look at this PR.

**Comment by skull132** [comment]
Unnaddressed changes required. Closing for now. Reopen once addressed.

**Pull request closed without merging**